### PR TITLE
Make sparkline tooltips overflowing (svg scope)

### PR DESCRIPTION
**Pull request by jotak**
Also fix an issue with the tooltip arrow not pointing exactly to the hovered datapoint

Fixes https://github.com/kiali/kiali/issues/2068

### CODE REVIEW

**Comment by jotak** [comment]
Capture d’écran de 2020-01-08 14-35-44

File: src/components/Charts/Charts.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.pf-c-chart svg:not(:root) {
+  overflow: visible;
+}
```
Comment (israel-hdez) [inline src/components/Charts/Charts.scss]: I'd prefer:
```css
.pf-c-chart svg {
￼  overflow: visible !important;
}
```

instead of using those weird pseudo-classes. Not a blocker, so you may merge without this change.

Comment (mtho11) [reply src/components/Charts/Charts.scss]: I did not personally test out @israel-hdez preference here but it does make sense if there are no other side effects.

Comment (jotak) [reply src/components/Charts/Charts.scss]: TBH I'm far from being a CSS expert - but I found this pattern `svg:not(:root)` on several places (I know, it's copy-paste (or stack-overflow) -based development) but also in our own codebase: https://github.com/kiali/kiali-ui/blob/master/src/components/Health/Health.scss (and I'm sure it's not me who wrote that).

So, given this, do you maintain this should be changed or not? I let you judge @mtho11 @israel-hdez

Comment (israel-hdez) [reply src/components/Charts/Charts.scss]: > ( ~and I'm sure it's not me who wrote that~ - actually this IS me who wrote that ... lol).
>

LOL!
 
> So, given this, do you maintain this should be changed or not? I let you judge @mtho11 @israel-hdez

Yes, I still prefer `!important` instead of the pseudo-classes.

To give a little explanation, I think you know, CSS have a _cascading_ nature. So, if you have two identical selectors that match, the "last" one is applied, for example, in:

```css
svg { display: none }
[...more css...]
svg { display: block }
```

`display: block` will be applied. Also, if more than one class matches, the more specific one is applied. So, if you have:

```css
a:hover { color: red }
a { color: blue }
```

You will see color red when you hover on a link regardless of the order in the CSS file, because `a:hover` is more specific. This is the rule being applied with those weird pseudo-classes, so if the generated CSS was like:

```css
.pf-c-chart svg:not(:root) { overflow: visible; }
.pf-c-chart svg { overflow: hidden; }
```

basically both selectors will _always_ match because an `svg` node can never be the _root_ element (in an HTML doc, only the `<html>` tag matches against CSS `:root`). But since you added `:not(:root)` you are turning it to a _more specific selector_ than the simple `svg`. Thus, the `overflow: visible` property is applied regardless of the order.

So, given that you just want to give more priority to the style, I'd lean towards using `!important` because it's clearer.

Comment (israel-hdez) [reply src/components/Charts/Charts.scss]: But as @mtho11 said, check that there are no side effects when using `!important` :)

Comment (jotak) [reply src/components/Charts/Charts.scss]: So after some tests, your suggestion works, but I had some doubt because the "not root" trick is actually already defined somewhere to hide overflow: `svg:not(:root) { overflow: hidden; }` (but without any class specifier). Not sure where that's defined, patternfly or victory. Anyway, it seems that "!important" wins!

**Comment by jotak** [comment]
@mattmahoneyrh sorry, I'm not sure to understand what you mean. Don't you have the tooltip updated when you move the cursor over different datapoints? You have to go out & in?

**Comment by jotak** [comment]
I'm merging this but @mattmahoneyrh please let me know for any follow-up
